### PR TITLE
Added custom markdown templates for feature_request, bug report and question

### LINKED ISSUE #12: Add Custom Bug Report + Feature Request Template
Create/Setup a custom template for 
- Reporting bugs
- Feature request

Resource: 
https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository

### PR DESCRIPTION
**Pull request by vishalbrlife**
## Description

Fixes: Issue #12

#### Purpose / goal

- The Purpose of this PR is to make the lives of the reporter easier by automating the pull request, bug reporting etc documentation.

#### Key changes
- [x] Added `bug_report.md`
- [x] Added `feature_request.md`
- [x] Added `pull_request.md`

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/pull_request.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (hawschiat) [inline .github/ISSUE_TEMPLATE/pull_request.md]: I think pull request templates should be under `PULL_REQUEST_TEMPLATE` directory or just named as so under `.github`

```
.github/PULL_REQUEST_TEMPLATE/pull_request.md
.github/PULL_REQUEST_TEMPLATE.md
```

The top-level metadata syntax is probably also not needed

File: .github/ISSUE_TEMPLATE/pull_request.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+about: Open a Pull Request for this project
+title: ''
+labels: 'type: pull request'
+assignees: ''
+
+---
+
+## Description
+
+Fixes issue: ___
```
Comment (hawschiat) [inline .github/ISSUE_TEMPLATE/pull_request.md]:
```suggestion
<!-- Link an issue to this PR. If none, this line can be removed -->
Fixes ___
```
Based on [GitHub docs](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+name: Bug Report
+about: Report Bugs
+title: ''
```
Comment (hawschiat) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
title: '[Bug] <title>'
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+labels: 'type: bug'
```
Comment (hawschiat) [inline .github/ISSUE_TEMPLATE/bug_report.md]:
```suggestion
labels: bug
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+name: Feature Request
+about: Please suggest a feature for this project
+title: ''
+labels: 'type: feature request'
```
Comment (hawschiat) [inline .github/ISSUE_TEMPLATE/feature_request.md]:
```suggestion
title: '[Feature Request] <title>'
labels: 'feature request'
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+assignees: ''
+
+---
+
+**Problems**
+
+**Screenshots/Demo**
```
Comment (hawschiat) [inline .github/ISSUE_TEMPLATE/bug_report.md]: I strongly suggest we also add sections as shown in this [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms#markdown-issue-template). It'll guide the reporter to write down as much as detail as possible 😃

File: .github/ISSUE_TEMPLATE/pull_request.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+<!-- What is the reason of this PR? -->
+
+...
+
+#### Key changes
+
+<!-- List the specific changes to guide reviewers on what to look for. -->
+
+- [x] ...
```
Comment (hawschiat) [inline .github/ISSUE_TEMPLATE/pull_request.md]:
```suggestion
- [x] ...

## Checklists

<!-- Make sure the following steps, where applicable, have been completed. You may add more items to the checklist as needed. -->

- [ ] Wrote unit tests
- [ ] Wrote API documentation
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
+---
+name: Pull Request
+about: Open a Pull Request for this project
+title: ''
+labels: 'type: pull request'
+assignees: ''
+---
+
```
Comment (hawschiat) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
```
I believe these aren't needed for the PR template

Comment (vishalbrlife) [reply .github/PULL_REQUEST_TEMPLATE.md]: Sorry, I missed your earlier comment!

**Review by hawschiat — APPROVED** [review]

**Pull request merged**